src/Castle.Core.Tests/TestDictionaryValidator.cs

```diff
@@ -47,11 +47,11 @@ public bool IsValid(IDictionaryAdapter dictionaryAdapter)
 		public string Validate(IDictionaryAdapter dictionaryAdapter)
 		{
 			List<String> errors = new List<string>();
-			var globalRules = AttributesUtil.GetTypeAttributes<IValidationRule>(dictionaryAdapter.Meta.Type);
+			var globalRules = AttributesUtil.GetTypeAttributes<ValidationRuleAttribute>(dictionaryAdapter.Meta.Type);

```

Comment:
Yes it's related to your change.  Before the change, the extension method 
```csharp
public static T[] GetAttributes<T>(this ICustomAttributeProvider member)
```
is on `ICustomAttributeProvider`, which is not available in .Net Core.  `ICustomAttributeProvider.GetCustomAttributes()` allows attributes that are not derived from `Attribute` (not CLS-Compliant).  Do you know whether this feature is needed?

After the change, the original test throws

```
Test Name:	DictionaryAdapterFactoryTestCase.CanValidateAndObtainDataErrorInformation
Test FullName:	Castle.Components.DictionaryAdapter.Tests.DictionaryAdapterFactoryTestCase.CanValidateAndObtainDataErrorInformation
Test Source:	E:\github\jeremymeng\Core\src\Castle.Core.Tests\Components.DictionaryAdapter.Tests\DictionaryAdapterFactoryTestCase.cs : line 1130
Test Outcome:	Failed
Test Duration:	0:00:00.028

Result StackTrace:	
at System.Attribute.GetCustomAttributes(MemberInfo element, Type type, Boolean inherit)
   at Castle.Core.Internal.AttributesUtil.GetAttributes[T](MemberInfo member) in E:\github\jeremymeng\Core\src\Castle.Core\Core\Internal\AttributesUtil.cs:line 69
   at Castle.Components.DictionaryAdapter.Tests.TestDictionaryValidator.Validate(IDictionaryAdapter dictionaryAdapter) in E:\github\jeremymeng\Core\src\Castle.Core.Tests\TestDictionaryValidator.cs:line 52
   at Castle.Components.DictionaryAdapter.Tests.TestDictionaryValidator.IsValid(IDictionaryAdapter dictionaryAdapter) in E:\github\jeremymeng\Core\src\Castle.Core.Tests\TestDictionaryValidator.cs:line 44
   at Castle.Components.DictionaryAdapter.DictionaryAdapterBase.<get_IsValid>b__87_0(IDictionaryValidator v) in E:\github\jeremymeng\Core\src\Castle.Core\Components.DictionaryAdapter\DictionaryAdapterBase.Validate.cs:line 33
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Castle.Components.DictionaryAdapter.DictionaryAdapterBase.get_IsValid() in E:\github\jeremymeng\Core\src\Castle.Core\Components.DictionaryAdapter\DictionaryAdapterBase.Validate.cs:line 33
   at Castle.Components.DictionaryAdapter.Tests.DictionaryAdapterFactoryTestCase.CanValidateAndObtainDataErrorInformation() in E:\github\jeremymeng\Core\src\Castle.Core.Tests\Components.DictionaryAdapter.Tests\DictionaryAdapterFactoryTestCase.cs:line 1134
Result Message:	Type passed in must be derived from System.Attribute or System.Attribute itself.
```